### PR TITLE
Add external location and storage credential services

### PR DESCRIPTION
**Pull request by vikrantpuppala**
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Add external location and storage credential services

### CODE REVIEW

File: server/src/test/java/io/unitycatalog/server/base/externallocation/BaseExternalLocationCRUDTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    CreateExternalLocation createExternalLocation =
+        new CreateExternalLocation()
+            .name(EXTERNAL_LOCATION_NAME)
+            .comment(COMMENT)
+            .url(URL)
+            .credentialName(CREDENTIAL_NAME);
+    // Fails as the credential does not exist
+    assertThatThrownBy(
+            () -> externalLocationOperations.createExternalLocation(createExternalLocation))
+        .isInstanceOf(ApiException.class);
```
Comment (tdas) [inline server/src/test/java/io/unitycatalog/server/base/externallocation/BaseExternalLocationCRUDTest.java]: can we not test out the exception message. so that we are sure that we are throwign the right error that its indeed because creds dont exist.

File: server/src/test/java/io/unitycatalog/server/base/externallocation/BaseExternalLocationCRUDTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    ExternalLocationInfo externalLocationInfo =
+        externalLocationOperations.createExternalLocation(createExternalLocation);
+    assertThat(externalLocationInfo.getName()).isEqualTo(EXTERNAL_LOCATION_NAME);
+    assertThat(externalLocationInfo.getComment()).isEqualTo(COMMENT);
+    assertThat(externalLocationInfo.getUrl()).isEqualTo(URL);
+    assertThat(externalLocationInfo.getCredentialId()).isEqualTo(storageCredentialInfo.getId());
+
+    // List external locations
+    assertThat(externalLocationOperations.listExternalLocations(Optional.empty()))
```
Comment (tdas) [inline server/src/test/java/io/unitycatalog/server/base/externallocation/BaseExternalLocationCRUDTest.java:79]: also test listing before adding the new location to make sure the result is empty.
